### DISCUSSION TITLE
How to print a label from a plugin

### DISCUSSION THREAD

**Discussion by SergeoLacruz** [inventree/InvenTree#8521] [General]
I create try to write a plugin using the NavigationMixin. How can i print a label from the plugin. I know the template and the printer driver but I cannot find what to call for printing. Thanks in advance.

**Comment by SchrodingersGat** [comment]
Sorry @SergeoLacruz there's not enough information to go on here. What do you mean printing from a navigationmixin?

**Comment by SergeoLacruz** [comment]
A bit more background, userstory :-)
I have a stockitem open and want to print a stockitem label. This is seven clicks. :-(
Ideal solution:
![image](https://github.com/user-attachments/assets/92ec9f79-9afa-4725-9eec-ac89da3c5789)
Place a button beside the others that prints a label using a predefined template on a predefined printer. One click
I think this is not possible using the plugin system. So I try to create a button there:
![image](https://github.com/user-attachments/assets/ffbc735e-92e5-4220-a319-b62b5e2a346a)
This is easy. But I need a way to trigger a print.

**Reply by SchrodingersGat** [reply]
Ah, right. Easiest method is to submit an API request to the server - and then look at the response, grab the reference to the printed label (if it is a pdf file it will be stored under `/media/`) and then instruct the browser to download that file

Check out how this is achieved in the UI currently:

https://github.com/inventree/InvenTree/blob/ad6f8596ebbeb69b406feda0addd9610bde35ef3/src/backend/InvenTree/templates/js/translated/label.js#L42
